azkaban-common/src/main/java/azkaban/server/HttpRequestUtils.java

```diff
@@ -159,6 +167,71 @@ public static ExecutionOptions parseFlowOptions(final HttpServletRequest req, fi
     return execOptions;
   }
 
+  /**
+   * Validate the ExecutionOption's Flow-Parameters against the application level Properties
+   * @throws ServletException if any of the parameter is invalid
+   */
+  public static void validatePreprocessFlowParameters(ExecutionOptions options, Props azProps)
+      throws ServletException {
+    List<String> errMsg = new ArrayList<>();
+
+    Map<String, String> flowParameters = options.getFlowParameters();
+    if (flowParameters == null || flowParameters.isEmpty()) {
+      return;
+    }
+    // if ALLOW_RESTART_ON_EXECUTION_STOPPED then ALLOW_RESTART_ON_STATUS += "EXECUTION_STOPPED,"
+    if (Boolean.parseBoolean(
+        flowParameters.getOrDefault(
+            FlowParameters.FLOW_PARAM_ALLOW_RESTART_ON_EXECUTION_STOPPED, "false"))
+    ){
+      flowParameters.put(
+          FlowParameters.FLOW_PARAM_ALLOW_RESTART_ON_STATUS,
+          Status.EXECUTION_STOPPED.name() + "," +
+              flowParameters.getOrDefault(FlowParameters.FLOW_PARAM_ALLOW_RESTART_ON_STATUS, "")
+      );
+      flowParameters.remove(FlowParameters.FLOW_PARAM_ALLOW_RESTART_ON_EXECUTION_STOPPED);
```

Comment:
Newly added, convert "allow.restart.on.execution.stopped" to "flow.retry.statuses"